src/Sound/SoundHandler.cpp

```diff
@@ -0,0 +1,216 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2020 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "SoundHandler.h"
+#include "3D/Camera.h"
+#include "Game.h"
+#include "Math.h"
+
+#include <fstream>
+#include <vector>
+
+namespace openblack::audio
+{
+
+SoundHandler::SoundHandler(std::unique_ptr<AudioDecoder>&& decoder, std::unique_ptr<AudioPlayer>&& player)
+    : _audioDecoder(std::move(decoder))
+    , _audioPlayer(std::move(player))
```

Comment:
error: ‘openblack::audio::SoundHandler::_audioDecoder’ will be initialized after ‘openblack::audio::SoundHandler::_audioPlayer’